mllib/src/main/scala/org/apache/spark/ml/image/ImageSchema.scala

```diff
@@ -0,0 +1,236 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml.image
+
+import java.awt.Color
+import java.awt.color.ColorSpace
+import java.io.ByteArrayInputStream
+import javax.imageio.ImageIO
+
+import scala.collection.JavaConverters._
+
+import org.apache.spark.annotation.{Experimental, Since}
+import org.apache.spark.input.PortableDataStream
+import org.apache.spark.sql.{DataFrame, Row, SparkSession}
+import org.apache.spark.sql.types._
+
+@Experimental
+@Since("2.3.0")
+object ImageSchema {
+
+  val undefinedImageType = "Undefined"
+
+  val imageFields: Array[String] = Array("origin", "height", "width", "nChannels", "mode", "data")
+
+  val ocvTypes: Map[String, Int] = Map(
```

Comment:
I assume we are going to expose this API as Java too? If so, I guess we should actually Java's map .. ?

Reply:
Does anyome have an idea about this please maybe?

Reply:
that's an interesting thought.  We want to expose it in both Java and Scala.  Worst case scenario, java users can use _ocvTypes variable - but if there is a better way to expose it, we should do it.

Reply:
Quick workaround I could think is, something like naming _ocvTypes to ocvTypesAsJava and documenting that's Java specific but.. want to know what others think ..

Reply:
There isn't a great option for Scala- and Java-friendly maps.  So far, we tend to prefix with "java" as in "javaOcvTypes"